daprdocs/content/en/reference/api/configuration_api.md

```diff
@@ -0,0 +1,243 @@
+
+```
+GET http://localhost:<daprPort>/v1.0/configuration/<storename>/subscribe
+```
+
+#### URL Parameters
+
+Parameter | Description
+--------- | -----------
+`daprPort` | The Dapr port
+`storename` | The `metadata.name` field in the user-configured `configuration.yaml` component file. Refer to the [Dapr configuration store component file structure](#component-file) mentioned above.
+
+#### Query Parameters
+
+If no query parameters are provided, all configuration items will subscribed to.
+To specifiy the keys of the configuration items to subscribe to, use one or more `key` query parameters. For example:
+
+```
+GET http://localhost:3500/v1.0/configuration/mystore/subscribe?key=config1&key=config2
+```
+
+#### Request Body
+
+None
+
+### HTTP Response
+
+#### Response Codes
+
+Code | Description
+---- | -----------
+`200`  | Subscribe operation successful
+`400`  | Configuration store is missing or misconfigured or malformed request
+`500`  | Failed to subscribe to configuration changes
+
+#### Response Body
+
+JSON-encoded value
+
+### Example
+
+```shell
+curl -X GET 'http://localhost:3500/v1.0-alpha1/configuration/subscribe?key=myConfigKey' 
+```
+
+> The above command returns the following JSON:
+
+```json
+{
+  "id": "<unique-id>"
+}
+```
+
+The returned `id` parameter can be used to unsubscribe to the keys provided on the API call.
+
+## Unsubscribe Configuration
+
+This endpoint lets you unsubscribe to configuration changes.
+
+### HTTP Request
+
+```
+GET http://localhost:<daprPort>/v1.0/configuration/<subscription-id>/unsubscribe
+```
+
+#### URL Parameters
+
+Parameter | Description
+--------- | -----------
+`daprPort` | The Dapr port
+`subscription-id` | The value from the `id` field returned from the response of the subscribe endpoint
+
+#### Query Parameters
+
+None
+
+#### Request Body
+
+None
+
+### HTTP Response
+
+#### Response Codes
+
+Code | Description
+---- | -----------
+`200`  | Unsubscribe operation successful
+`400`  | Configuration store is missing or misconfigured or malformed request
+`500`  | Failed to unsubscribe to configuration changes
+
+#### Response Body
+
+None
+
+### Example
+
+```shell
+curl -X GET 'http://localhost:3500/v1.0-alpha1/configuration/bf3aa454-312d-403c-af95-6dec65058fa2/unsubscribe' 
+```
+
+## Optional Application (User Code) Routes
+
+### Provide a route for Dapr to send configuration changes
+
+When subscribing to configuration changes, Dapr will invoke the application whenever a configuration item changes.
+
+#### HTTP Request
+
+```
+POST http://localhost:<appPort>/configuration/<store-name>/<key>
+```
+
+#### URL Parameters
+
+Parameter | Description
+--------- | -----------
+`appPort` | The application port
+`storename` | The `metadata.name` field in the user-configured `configuration.yaml` component file. Refer to the [Dapr configuration store component file structure](#component-file) mentioned above
+`key` | The key of the changed configuration item
+
+#### Request Body
```

Comment:
What is the example showing here? is this supposed to show the routes on the application?

Reply:
Yes, we do the exact same thing on the pub/sub API reference.

Reply:
If you are referring to this https://docs.dapr.io/reference/api/pubsub_api/#http-response-body then that is not what is shown below. What is `version`. And I do not understand how is this used? Whereas subscription for pub/sub returns a list of the routes, from the app, what is this calling on the app? Please explain.

Reply:
No, it's referring to https://docs.dapr.io/reference/api/pubsub_api/#provide-routes-for-dapr-to-deliver-topic-events.

Reply:
Question: Like pub/sub are there any success/failure codes that the application can return to Dapr?

Reply:
No